cabal-install/Distribution/Client/Dependency/Modular/Message.hs

```diff
@@ -47,6 +47,9 @@ showMessages p sl = go [] 0
     go v l (TryF qfn b : Enter : Failure c fr : Leave : ms) = (atLevel (add (F qfn) v) l $ "rejecting: " ++ showQFNBool qfn b ++ showFR c fr) (go v l ms)
     go v l (TryS qsn b : Enter : Failure c fr : Leave : ms) = (atLevel (add (S qsn) v) l $ "rejecting: " ++ showQSNBool qsn b ++ showFR c fr) (go v l ms)
     go v l (Next (Goal (P qpn) gr) : TryP qpn' i : ms@(Enter : Next _ : _)) = (atLevel (add (P qpn) v) l $ "trying: " ++ showQPNPOpt qpn' i ++ showGRs gr) (go (add (P qpn) v) l ms)
+    go v l (Next (Goal (P qpn) gr) : Failure c fr : ms) =
+        let v' = add (P qpn) v
+        in (atLevel v' l $ showPackageGoal qpn gr) $ (atLevel v' l $ showFailure c fr) (go v l ms)
```

Comment:
@kosmikus Thanks for reviewing this.

I assumed that when `go` is called with `Next (Goal (P qpn) _) : _`, the following messages also relate to `qpn`, and it would make sense for `P qpn` to be added to the stack of variables.  However, there isn't an easy way to determine when to pop the variable.  I instead tried to replicate that behavior by adding a special case for a failure after a package goal choice.  Since `Next` is only followed by `Failure` or `TryP`, and the `TryP` case already adds the right variable to the stack, I thought that my change would handle the remaining cases.

It seems like `BuildFailureNotInIndex` is the only `FailReason` that should come directly after a package goal choice, so I think that there is no difference in behavior between my PR and your suggestion.  I would still need to match on the message before `BuildFailureNotInIndex`, because `BuildFailureNotInIndex` contains a `PN`, and the stack needs a `QPN`.  Would you like me to make that change?